src/main/java/com/bookstore/controller/publisher/PublisherController.java

```diff
@@ -0,0 +1,110 @@
+package com.bookstore.controller.publisher;
+import com.bookstore.jpa.book.BookRepository;
+import com.bookstore.jpa.publisher.Publisher;
+import com.bookstore.jpa.publisher.PublisherRepository;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.stereotype.Controller;
+import org.springframework.ui.Model;
+import org.springframework.web.bind.annotation.*;
+
+import java.util.ArrayList;
+import java.util.Collection;
+
+
+/**
+ * The Publisher Rest API Controller
+ * @author Muneeb Nasir
+ * @version 4806.1
+ */
+@Controller
+public class PublisherController {
+
+    @Autowired
+    private BookRepository repository;
+
+    @Autowired
+    private PublisherRepository publisherRepo;
+
+    /**
+     * Getter for all publishers in the Database
+     * @param model
+     * @return Thymeleaf Page
+     */
+    @GetMapping(path = "/AmazinBookStore-publishers")
+    public String addressBookView(Model model) {
+
+        Collection<Publisher> publishers = publisherRepo.findAll();
+        model.addAttribute("allPublishers", publishers);
+
+        // Empty Publisher Object - namespace
+        model.addAttribute("newPublisher", new Publisher());
+        return "AmazinBookStore-publishers";
+    }
+
+    @PostMapping(path = "/AmazinBookStore-addNewPublisher", consumes = "application/json")
+    @ResponseBody
+    public ResponseEntity<HttpStatus> addNewPublisher(@RequestBody Publisher publisher) {
+
+        Publisher newPublisher = new Publisher(publisher.getName(), publisher.getLocation());
+        publisherRepo.save(newPublisher);
+
+        return new ResponseEntity<>(HttpStatus.OK);
+    }
+
+    @GetMapping(path = "/AmazinBookStore-AllIDs", produces = "application/json")
+    @ResponseBody
+    public ResponseEntity<Collection> all() {
+        ArrayList publisherCollection = new ArrayList<>();
+        for (Publisher pub: publisherRepo.findAll()){
+            publisherCollection.add(pub.getId());
+        }
+
+        return new ResponseEntity<>(publisherCollection, HttpStatus.OK);
+    }
+
+    @GetMapping(path = "/AmazinBookStore-publishersViewAll", produces = "application/json")
+    @ResponseBody
+    public ResponseEntity<Collection> addressBooksViewAll() {
+
+        Collection<Publisher> publishers = publisherRepo.findAll();
+        return new ResponseEntity<>(publishers, HttpStatus.OK);
+    }
+
+    @PostMapping(path = "/AmazinBookStore-updatePublisher/{id}", consumes = "application/json")
+    @ResponseBody
+    public ResponseEntity<Publisher> updatePublisher(@RequestBody Publisher publisher,
+                                                     @PathVariable("id") long id) {
+
+        Publisher tempPublisher = publisherRepo.findById(id);
+        if (tempPublisher == null){
+            return new ResponseEntity<>(null, HttpStatus.BAD_REQUEST);
```

Comment:
Status code 400 is used when server cannot process the request due to mistake made by the user like using wrong syntax, so it would be inappropriate to use it here. I think returning 200 with empty object or maybe a 204 (No Content) status code.